src/client/elm/Transaction/Event.elm

```diff
@@ -552,6 +569,54 @@ protocolUpdateDecoder =
         |> required "specificationAuxiliaryData" D.string
 
 
+decoderFromMaybe : String -> Maybe a -> D.Decoder a
+decoderFromMaybe error m =
+    case m of
+        Just n ->
+            D.succeed n
+
+        Nothing ->
+            D.fail error
+
+
+hexDecoder : D.Decoder B.Bytes
+hexDecoder =
+    D.string |> D.andThen (\str -> decoderFromMaybe "Failed to decode Hex string" (Hex.Convert.toBytes str))
+
+
+cborDecoder : Cbor.Decode.Decoder a -> B.Bytes -> D.Decoder a
+cborDecoder decoder bytes =
+    decoderFromMaybe "Failed to decode Cbor" <| Cbor.Decode.decode decoder bytes
+
+
+memoDecoder : D.Decoder Memo
```

Comment:
Please document the behaviour of this function and the rationale for it. Why does  it do this ad-hoc parsing? It will not be obvious in the future.